src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java

```diff
@@ -121,30 +128,51 @@ public CreateDeploymentResult createDeploymentEvent(
         request.addHeader(getBasicAuthHeader(restApiKey));
         request.addHeader(JSON_TYPE_HEADER);
 
-        return parseCreateDeploymentEventResponse(httpClient.execute(request));
+        return parseApiResult(httpClient.execute(request), CreateDeploymentResult.class);
     }
 
     @Override
     public ExecutionResult getExecutionResults(String eventId) throws IOException, MablSystemError {
-
         final String url = restApiBaseUrl + String.format(DEPLOYMENT_RESULT_ENDPOINT_TEMPLATE, eventId);
-        final HttpGet request = new HttpGet(url);
+        return parseApiResult(httpClient.execute(buildGetRequest(url)), ExecutionResult.class);
+    }
 
-        request.addHeader(getBasicAuthHeader(restApiKey));
+    public GetApiKeyResult getApiKeyResult(String formApiKey) throws IOException, MablSystemError {
+        final String url = restApiBaseUrl + String.format(GET_ORGANIZATION_ENDPOINT_TEMPLATE, formApiKey);
+        return parseApiResult(httpClient.execute(buildGetRequest(url)), GetApiKeyResult.class);
+    }
 
-        return parseExecutionResultResponse(httpClient.execute(request));
+    public GetApplicationsResult getApplicationsResult(String organizationId) throws IOException, MablSystemError {
+        final String url = restApiBaseUrl + String.format(GET_APPLICATIONS_ENDPOINT_TEMPLATE, organizationId);
+        return parseApiResult(httpClient.execute(buildGetRequest(url)), GetApplicationsResult.class);
     }
 
-    private ExecutionResult parseExecutionResultResponse(final HttpResponse response) throws IOException, MablSystemError {
+    public GetEnvironmentsResult getEnvironmentsResult(String organizationId) throws IOException, MablSystemError {
+        final String url = restApiBaseUrl + String.format(GET_ENVIRONMENTS_ENDPOINT_TEMPLATE, organizationId);
+        return parseApiResult(httpClient.execute(buildGetRequest(url)), GetEnvironmentsResult.class);
+    }
+
+    private HttpGet buildGetRequest(String url) throws MablSystemError {
+        try {
+            final HttpGet request = new HttpGet(url);
+            request.addHeader(getBasicAuthHeader(restApiKey));
+            return request;
+        } catch (IllegalArgumentException e) {
+            throw new MablSystemError(String.format("Unexpecetd status from mabl trying to build API url: %s", url));
```

Comment:
Typo: `Unexpeceted`

Reply:
Also, further down `API on execution result fetch` message should be updated, since the parse method is now generic and not just executions.